skglm/datafits/single_task.py

```diff
@@ -107,6 +108,101 @@ def full_grad_sparse(
     def intercept_update_step(self, y, Xw):
         return np.mean(Xw - y)
 
+class CensoredQuadratic(Quadratic):
+    """Quadratic datafit with access to Xty but not to y.
+
+    The datafit reads:
+
+    .. math:: 1 / (2 xx  n_"samples") ||Xw||_2 ^ 2 - 1 / (n_"samples") w^\top (X^\top y)
+
+    Attributes
+    ----------
+    Xty : array, shape (n_features,)
+        Pre-computed quantity used during the gradient evaluation.
+        Equal to ``X.T @ y``.
+
+    Note
+    ----
+    The class is jit compiled at fit time using Numba compiler.
+    This allows for faster computations.
+    """
+
+    def __init__(self, Xty, y_mean):
+        self.Xty = Xty
+        self.y_mean = y_mean
```

Comment:
We better leave the constructor to define the Datafit hyper-parameters.
We can move `Xty` and `y_mean` to the `initialize` method

Reply:
this would break the existing code: all solvers call `datafit.initialize(X, y)` internally

Reply:
maybe a better API would be to instantiate all datafits with X, y and whatever they need (each has its own API)
then call `datafit.initialize()` that would use all stored attributes

eg
```python
datafit = Quadratic(X, y)
penalty = L1(alpha=0.1)
solver = AndersonCD()
solver.solve(datafit, penalty)
```

It would give more freedom to each datafit, to require various quantities. 

To me this makes more sense because we have datafits like Cox depneding on more than just X and y.

Wdyt @BadrMoufad ?